PWGCF/EbyEFluctuations/Tasks/netchargeFluctuations.cxx

```diff
@@ -9,94 +9,631 @@
+      histogramRegistry.fill(HIST("hDcaXY_before"), track.dcaXY());
+      histogramRegistry.fill(HIST("hDcaZ_before"), track.dcaZ());
+      histogramRegistry.fill(HIST("hPtDcaXY_before"), track.pt(), track.dcaXY());
+      histogramRegistry.fill(HIST("hPtDcaZ_before"), track.pt(), track.dcaZ());
+
+      if (!selTrack(track)) {
         continue;
-      if (!(track.pt() > 0.2 && track.pt() < 2.))
+      }
+      if (track.sign() == 0)
         continue;
+      histogramRegistry.fill(HIST("hDcaXY_after"), track.dcaXY());
+      histogramRegistry.fill(HIST("hDcaZ_after"), track.dcaZ());
+      histogramRegistry.fill(HIST("hPt"), track.pt());
+      histogramRegistry.fill(HIST("hEta"), track.eta());
+      histogramRegistry.fill(HIST("hPtDcaXY_after"), track.pt(), track.dcaXY());
+      histogramRegistry.fill(HIST("hPtDcaZ_after"), track.pt(), track.dcaZ());
+      histogramRegistry.fill(HIST("hTPCCrossedrows_after"), track.tpcNClsCrossedRows());
+      histogramRegistry.fill(HIST("hTPCchi2perCluster_after"), track.tpcChi2NCl());
+      histogramRegistry.fill(HIST("hITSchi2perCluster_after"), track.itsChi2NCl());
+
+      if (track.sign() == 1) {
+        fpos += 1;
+        termp = fpos * (fpos - 1);
+      }
+
+      if (track.sign() == -1) {
+        fneg += 1;
+        termn = fneg * (fneg - 1);
+      }
 
-      histos.fill(HIST("hDCAxy_bef"), track.dcaXY());
-      histos.fill(HIST("hDCAz_bef"), track.dcaZ());
+      posneg = fpos * fneg;
+      netCharge(fpos, fneg, fpos * fpos, fneg * fneg, termp, termn, posneg, cent);
+    } // tracks
+
+    return;
+  }
+
+  template <RunType run, typename C, typename T, typename M, typename P>
+  void histosMcRecoGen(C const& coll, T const& inputTracks, M const& mcCollisions, P const& mcParticles)
+  {
+    (void)mcCollisions;
+    if (!coll.has_mcCollision()) {
+      return;
+    }
+
+    histogramRegistry.fill(HIST("gen_hVtxZ_before"), coll.mcCollision().posZ());
+    histogramRegistry.fill(HIST("rec_hVtxZ_before"), coll.posZ());
+
+    if (cNoItsROBorder && !coll.selection_bit(aod::evsel::kNoITSROFrameBorder)) {
+      return;
+    }
+    if (cTFBorder && !coll.selection_bit(aod::evsel::kNoTimeFrameBorder)) {
+      return;
+    }
+    if (cPileupReject && !coll.selection_bit(aod::evsel::kNoSameBunchPileup)) {
+      return;
+    }
+    if (cZVtxTimeDiff && !coll.selection_bit(aod::evsel::kIsGoodZvtxFT0vsPV)) {
+      return;
+    }
+    if (cItsTpcVtx && !coll.selection_bit(aod::evsel::kIsVertexITSTPC)) {
+      return;
+    }
 
+    if (std::abs(coll.posZ()) > vertexZcut) {
+      return;
+    }
+
+    if constexpr (run == kRun3) {
+      if (cSel8Trig && !coll.sel8()) {
+        return;
+      }
+
+      cent = coll.centFT0M(); // centrality for run3
+      mult = coll.multFT0M();
+    } else {
+      if (cSel7Trig && !coll.sel7()) {
+        return;
+      }
+
+      cent = coll.centRun2V0M(); // centrality for run2
+      mult = coll.multFV0M();    // multiplicity for run2
+    }
```

Comment:
In here the code in `selCollision` is basically replicated
I would suggest not replicating code. Replicating code is quite error prone, difficult to maintain, and difficult to evolve